SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LocalReceptionistProxyServerManager.h"
+
+#include "Internationalization/Regex.h"
+#include "SpatialCommandUtils.h"
+#include "UObject/CoreNet.h"
+
+DEFINE_LOG_CATEGORY(LogLocalReceptionistProxyServerManager);
+
+#define LOCTEXT_NAMESPACE "FLocalReceptionistProxyServerManager"
+
+FLocalReceptionistProxyServerManager::FLocalReceptionistProxyServerManager()
+: RunningCloudDeploymentName("")
+{
+}
+
+bool FLocalReceptionistProxyServerManager::KillProcessBlockingPort()
```

Comment:
ohhh nice! does that take care of the proxy sticking around if you crash? 
could we log some information about which process we are killing? just in case it is actually a different process.

what if we process stays around, we restart unreal and then try to start a local deployment. will that work?

Reply:
Sure I will see if you can add some logging! At the moment, I am killing the process when we try to start a proxy and we can not. Local Deployment has a corresponding kill process function but I am not sure if and how is used. I will test it and take a look.

Reply:
Yep you were right I need to fix this if the engine crashes and we start a cloud deployment it will kill it but if we start a local one it wont. So I need to fix that

Reply:
I fixed this issue by adding an init function in the LocalReceptionistProxyServerManager that is invoked when we StartUp the ToolbarModule